src/main/kotlin/lotto/domain/LottoMachine.kt

```diff
@@ -0,0 +1,57 @@
+package lotto.domain
+
+import lotto.data.Lotto
+import lotto.data.LottoNumber
+import lotto.data.LottoRanking
+import lotto.data.LottoRanking.FifthPlace
+import lotto.data.LottoRanking.FirstPlace
+import lotto.data.LottoRanking.FourthPlace
+import lotto.data.LottoRanking.None
+import lotto.data.LottoRanking.SecondPlace
+import lotto.data.LottoRanking.ThirdPlace
+import lotto.data.WinningLotto
+
+object LottoMachine {
+
+    fun createSelectLotto(lottoNumbers: Set<LottoNumber>): Lotto {
+        return Lotto(lottoNumbers)
+    }
+
+    fun createWinningLotto(winningNumbers: List<Int>, bonusNumber: Int): WinningLotto {
+        val winningLotto = LottoNumber.createLottoNumbers(winningNumbers)
+        val bonusLottoNumber = LottoNumber.from(bonusNumber)
+        return WinningLotto(Lotto(winningLotto), bonusLottoNumber)
+    }
+
+    fun checkLotto(purchaseLotto: Lotto, winningLotto: WinningLotto): LottoRanking {
+        val intersectNumber = winningLotto.lotto.selectNumbers.intersect(purchaseLotto.selectNumbers)
+
+        return when (intersectNumber.size) {
+            FirstPlace.matchingNumberCnt -> FirstPlace
+            SecondPlace.matchingNumberCnt, ThirdPlace.matchingNumberCnt -> {
+                checkSecondPlace(purchaseLotto, winningLotto.bonusNumber)
+            }
+            FourthPlace.matchingNumberCnt -> FourthPlace
+            FifthPlace.matchingNumberCnt -> FifthPlace
+            else -> None
+        }
+    }
+
+    fun createWinningRate(cash: Int, winningStatus: Map<LottoRanking, Int>): Float {
+        val totalPrice = createTotalWinningPrice(winningStatus)
+
+        return totalPrice / cash.toFloat()
+    }
```

Comment:
전략 패턴(strategy pattern)을 활용한다면 다음과 같이 작성할 수 있을 것 같네요.
```suggestion
    fun createWinningRate(cash: Int, winningStatus: Map<LottoRanking, Int>): Float {
        return winningRateCalculator.calculate(info);
    }
```

이렇게 작성할 경우 나중에 기획에 따라 나라별로 이율 계산 방식니 달라진다고 해도 쉽게 변경할 수 있어요.

```kotlin
    fun createWinningRate(cash: Int, winningStatus: Map<LottoRanking, Int>): Float {
        return winningRateCalculators.calculate(info= info, country: getCountry());
    }
```

Reply:
당첨 번호와 통계에 대한 분석을 로또 머신에서 할 수 있긴 합니다. 
그런데 전략패턴을 사용한다고 로또 머신에서 통계등의 작업을 하지 않는게 아니에요. 

자동차를 예로 들어볼때 자동차안에서 핸들을 움직인다고 하고 엑셀을 밟고, 브레이크를 밟을때 자동차는 앞/뒤/좌/우로 움직이는데, 실질적으로 책임지는건 핸들이 아니라 핸들부터 각각의 모든 부품들이 각자 위치에서 자기 책임을 다 했을때 최종적으로 우기아 의도한대로 움직이게 되는 것이죠. 

그리고 로또 머신이 수익율 계산을 하는게 맞는가? 고민을 하신다면 로또 머신의 Actor에 대해 고민해보면 좋을 것 같아요.  로또 머신이 로또를 발급하고 비교하고 수익율을 계산하는 서비스 계층이라면 적절한 책임이고, 그게 아니라면 과도한 책임이겠죠